active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,572 @@
+
+<!-- If you're modifying an existing API, included a link here to the
+existing page(s) -->
+
+<!-- For example, this section is a place to explain why you're adding this API rather than
+modifying an existing API. -->
+
+<!-- For example, this is a place to provide a brief explanation of some dependent
+area, just explanation enough to understand this new API, rather than telling
+the reader "go read 100 pages of background information posted at ...". -->
+
+
+# Description
+
+Most applications that take user input need to have validation logic to ensure that the user has entered the expected information. The validation checks can be based on type, range, format, or other application-specific requirements. For example:
+![validation example](basicvalidationexample.png)
+
+
+# Examples
+
+## End to end input validation
+
+In this sample, we want to display a validation error on a textbox when user input is invalid. This is the recommended model for validating fields. The validation logic with DataAnnoations displayed in this sample is proper to managed apps. Native apps can use INotifyDataErrorInfo as described below and substitute their current validation logic in place of the .Net APIs.
+
+![first name](firstnameexample.png)
+
+Use INotifyDataErrorInfo with x:Bind to update your view. This API can be implemented by any class, and allows the object to raise an event if one of its properties is set to an invalid value. When the source of a Xaml x:Bind implements this interface, the x:Bind will put the validation errors onto the bind target. Xaml controls such as TextBox will recognize these errors and provide visualization for them.
+
+### XAML
+
+a user input control with the input bound:
+
+```xaml
+<TextBox Name="ValidationTextbox" 
+         Text="{x:Bind  ViewModel.Person.Name} "/>
+
+```
+#### Code - ViewModel
+
+The “ViewModel” property is the View Model for the Page on which this TextBox is displayed (this.ViewModel).
+``` csharp
+
+public class ViewModel
+{
+    public ViewModel()
+    {
+        Person = new Person();
+    }
+
+    public Person Person { get; set; }
+
+}
+```
+
+#### C# only - Person (Model)
+Person.Name is the source of the bind (this.ViewModel.Person.Name).  Person derives from ValidationWrapper (app code) which will provide the data validation for Name.  It’s marked with [MinLength] and [MaxLength] data annotations.  The property accessors use GetValue/SetValue in the base ValidationWrapper class for storage and to check the validity.
+
+``` csharp
+using System.ComponentModel.DataAnnotations;
+    
+public class Person : ValidationWrapper
+{
+    [MinLength(4, ErrorMessage = "Name must be at least 4 characters")]
+    [MaxLength(6, ErrorMessage = "Name must be no more than 6 characters")]
+    public string Name
+    {	
+        get { return GetValue<string>("Name"); }
+        set { SetValue("Name", value); }
+    }
+}
+```
+
+#### Code – INotifyDataErrorInfo base class
```

Comment:
Shouldn't this header be `NotifyDataErrorInfo` not `INotifyDataErrorInfo`?

Reply:
I think this should be "Sample INotifyDataErrorInfo implementation", as this base class won't be part of WinUI (See next comment from @LucasHaines)